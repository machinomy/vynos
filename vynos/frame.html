<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <link href="https://fonts.googleapis.com/css?family=Alegreya+Sans:400,500,700|Roboto|Source+Sans+Pro:400,600" rel="stylesheet">
    <style>
        #mount-point {
            width: 100%;
            height: 440px;
            overflow: hidden;
        }

        #mount-point #spinner {
            position: absolute;
            margin: auto;
            top: 0;
            left: 0;
            bottom: 0;
            right: 0;
            width: 150px;
            height: 150px;
            text-align: center;
        }

        #mount-point #spinner > * {
            margin: 0;
        }

        #mount-point #spinner .main-header svg {
            width: 100px;
        }

        #mount-point #spinner > .main-header {
            font: 400 40px 'Alegreya Sans', sans-serif !important;
            color: #D04071 !important;
            line-height: 0 !important;
        }

        #customNetwork {
          text-transform: none !important;
        }

        html, body {
            background: white;
        }
    </style>
</head>
<body>
    <div id="mount-point">
        <div id="spinner">
            <h1 class="main-header">
                <svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
                        viewBox="0 0 200 200" style="enable-background:new 0 0 200 200;" xml:space="preserve">
                    <style type="text/css">
                    .st0{fill:#D04071;}
                    </style>
                    <g transform="rotate(0 50 50)">
                        <path class="st0" d="M100,40L100,40c3.3,0,6,2.7,6,6v28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6V46C94,42.7,96.7,40,100,40z	">
                    </path>
                    </g>
                    <g transform="rotate(120 50 50)">
                        <path class="st0" d="M68.3-78.3L68.3-78.3c3.3,0,6,2.7,6,6v28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6l0-28
                            C62.3-75.6,65-78.3,68.3-78.3z	">
                        </path>
                    </g>
                    <g transform="rotate(240 50 50)">
                        <path class="st0" d="M-18.3,8.3L-18.3,8.3c3.3,0,6,2.7,6,6l0,28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6l0-28
                        C-24.3,11-21.6,8.3-18.3,8.3z	">
                        </path>
                    </g>
                </svg>
                vynos
            </h1>
        </div>
    </div>
    <script>
      var frameSrc = '/frame.js'
      var src = window.location.href.match(/dev=true/) ? frameSrc.replace('.js', '.dev.js') : frameSrc;
      var scriptTag = document.createElement('script')
      scriptTag.setAttribute('src', src)
      document.body.appendChild(scriptTag)
    </script>
</body>
</html>
