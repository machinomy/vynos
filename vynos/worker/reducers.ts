import {INITIAL_STATE, WorkerState} from "./WorkerState";
import {reducerWithInitialState, ReducerBuilder} from "typescript-fsa-reducers";
import * as actions from "./actions";

const reducers: ReducerBuilder<WorkerState, WorkerState> = reducerWithInitialState(INITIAL_STATE)
  .case(actions.setWallet, actions.setWalletHandler)
  .case(actions.setLastMicropaymentTime, actions.setLastMicropaymentTimeHandler)
  .case(actions.setKeyring, actions.setKeyringHandler)
  .case(actions.setDidStoreMnemonic, actions.setDidStoreMnemonicHandler)
  .case(actions.restoreWallet, actions.restoreWalletHandler)
  .case(actions.setTransactionPending, actions.setTransactionPendingHandler)
  .case(actions.rememberPage, actions.rememberPageHandler)
  .case(actions.setLastUpdateDb, actions.setLastUpdateDbHandler)
  .case(actions.setPreferences, actions.setPreferencesHandler)
  .case(actions.setPersistentState, actions.setPersistentStateHandler)

export default reducers
