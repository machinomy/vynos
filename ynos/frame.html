<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, minimum-scale=1, initial-scale=1, shrink-to-fit=no">
</head>
<body>
    <div id="mount-point">Loading...</div>
    <script src="/frame.bundle.js"></script>
</body>
</html>
