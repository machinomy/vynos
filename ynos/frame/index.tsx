/*
import * as React from "react";
import {connect} from "react-redux";
import {FrameState} from "../../state";
import TermsSubpage from "./TermsSubpage";
import PasswordSubpage from "./PasswordSubpage";
import MnemonicSubpage from "./MnemonicSubpage";

export interface InitPageProps {
  needAcceptTerms: boolean
  needSetPassword: boolean
}

const InitPage: React.SFC<InitPageProps> = (props) => {
  if (props.needAcceptTerms) {
    return <TermsSubpage />
  } else if (props.needSetPassword) {
    return <PasswordSubpage />
  } else {
    return <MnemonicSubpage />
  }
}

function mapStateToProps(state: FrameState): InitPageProps {
  return {
    needAcceptTerms: !state.temp.initPage.didAcceptTerms,
    needSetPassword: !state.temp.initPage.mnemonic,
  }
}

export default connect<InitPageProps, undefined, any>(mapStateToProps)(InitPage)
*/
