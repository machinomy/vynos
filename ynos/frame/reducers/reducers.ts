import * as redux from "redux";
import {FrameState, INITIAL_FRAME_STATE} from "./state";
import { reducerWithInitialState } from "typescript-fsa-reducers";
import actions from "../actions";
import {Reducer} from "redux";

const tempReducer = reducerWithInitialState(INITIAL_FRAME_STATE.temp)
  .case(actions.temp.setWorkerProxy, actions.temp.setWorkerProxyHandler)
  .case(actions.temp.init.didAcceptTerms, actions.temp.init.didAcceptTermsHandler)
  .case(actions.temp.init.didReceiveMnemonic, actions.temp.init.didReceiveMnemonicHandler)

const sharedReducer = reducerWithInitialState(INITIAL_FRAME_STATE.shared)
  .case(actions.shared.setSharedState, actions.shared.setSharedStateHandler)

const reducers: Reducer<FrameState> = redux.combineReducers({
  temp: tempReducer,
  shared: sharedReducer
});

export default reducers
