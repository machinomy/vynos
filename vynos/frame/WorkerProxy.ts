import StreamProvider from '../lib/StreamProvider'
import { EventEmitter } from 'events'
import { SharedState, Preferences } from '../worker/WorkerState'
import { JSONRPC, randomId } from '../lib/Payload'
import { isSharedStateBroadcast, SharedStateBroadcastType } from '../lib/rpc/SharedStateBroadcast'
import {
  DidStoreMnemonicRequest,
  GenKeyringRequest,
  GenKeyringResponse,
  GetSharedStateRequest,
  GetSharedStateResponse,
  LockWalletRequest,
  RestoreWalletRequest,
  RestoreWalletResponse,
  RememberPageRequest,
  UnlockWalletRequest,
  UnlockWalletResponse,
  TransactonResolved,
  ChangeNetworkRequest,
  GetPrivateKeyHexRequest,
  GetPrivateKeyHexResponse,
  SetPreferencesRequest,
  SetApproveByIdRequest,
  SetRejectByIdRequest,
  ClearTransactionMetastorageRequest,
  ClearReduxPersistentStorageRequest,
  ClearChannelMetastorageRequest,
  CloseChannelRequest,
  ListChannelsRequest,
  ListChannelsResponse,
  ClearMachinomyStorageRequest,
  ClearAccountInfoRequest
} from '../lib/rpc/yns'
import { Action } from 'redux'
import Web3 = require('web3')
import { PaymentChannel, PaymentChannelSerde } from 'machinomy/lib/PaymentChannel'

export default class WorkerProxy extends EventEmitter {
  provider: StreamProvider
  web3: Web3

  constructor () {
    super()
    this.provider = new StreamProvider('WorkerProxy')
    this.provider.listen(SharedStateBroadcastType, data => {
      if (isSharedStateBroadcast(data)) {
        this.emit(SharedStateBroadcastType, data)
      }
    })
    this.web3 = new Web3(this.provider)
  }

  getWeb3 (): Web3 {
    return new Web3(this.provider)
  }

  doLock (): Promise<void> {
    let request: LockWalletRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: LockWalletRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  doUnlock (password: string): Promise<string | undefined> {
    let request: UnlockWalletRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: UnlockWalletRequest.method,
      params: [password]
    }
    return this.provider.ask(request).then((response: UnlockWalletResponse) => {
      return response.error
    })
  }

  genKeyring (password: string): Promise<string> {
    let request: GenKeyringRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: GenKeyringRequest.method,
      params: [password]
    }
    return this.provider.ask(request).then((response: GenKeyringResponse) => {
      return response.result
    })
  }

  restoreWallet (password: string, type: string, value: string): Promise<string> {
    let request: RestoreWalletRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: RestoreWalletRequest.method,
      params: [password, type, value]
    }
    return this.provider.ask(request).then((response: RestoreWalletResponse) => {
      return response.result
    })
  }

  getSharedState (): Promise<SharedState> {
    let request: GetSharedStateRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: GetSharedStateRequest.method,
      params: []
    }
    return this.provider.ask(request).then((response: GetSharedStateResponse) => {
      return response.result
    })
  }

  didStoreMnemonic (): Promise<void> {
    let request: DidStoreMnemonicRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: DidStoreMnemonicRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  rememberPage (path: string): void {
    let request: RememberPageRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: RememberPageRequest.method,
      params: [path]
    }
    this.provider.ask(request).then(() => {
      // Do Nothing
    })
  }

  resolveTransaction (): void {
    let request: TransactonResolved = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: TransactonResolved.method,
      params: []
    }
    this.provider.ask(request).then(() => {
      // Do Nothing
    })
  }

  getPrivateKeyHex (): Promise<string> {
    let request: GetPrivateKeyHexRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: GetPrivateKeyHexRequest.method,
      params: []
    }
    return this.provider.ask(request).then((response: GetPrivateKeyHexResponse) => {
      return response.result
    })
  }

  dispatch<A extends Action> (action: A) {
    console.warn('WorkerProxy#dispatch', action)
  }

  changeNetwork (): Promise<void> {
    let request: ChangeNetworkRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: ChangeNetworkRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  setPreferences (preferences: Preferences): Promise<void> {
    let request: SetPreferencesRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: SetPreferencesRequest.method,
      params: [preferences]
    }

    return this.provider.ask(request).then(() => {
      return
    })
  }

  setApproveById (id: string): void {
    let request: SetApproveByIdRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: SetApproveByIdRequest.method,
      params: [id]
    }
    this.provider.ask(request).then(() => {
      // Do Nothing
    })
  }

  setRejectById (id: string): void {
    let request: SetRejectByIdRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: SetRejectByIdRequest.method,
      params: [id]
    }
    this.provider.ask(request).then(() => {
      // Do Nothing
    })
  }

  clearTransactionMetastorage (): Promise<void> {
    let request: ClearTransactionMetastorageRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: ClearTransactionMetastorageRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  clearChannelMetastorage (): Promise<void> {
    let request: ClearChannelMetastorageRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: ClearChannelMetastorageRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  clearReduxPersistentStorage (): Promise<void> {
    let request: ClearReduxPersistentStorageRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: ClearReduxPersistentStorageRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  clearMachinomyStorage (): Promise<void> {
    let request: ClearMachinomyStorageRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: ClearMachinomyStorageRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  clearAccountInfo (): Promise<void> {
    let request: ClearAccountInfoRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: ClearAccountInfoRequest.method,
      params: []
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  closeChannel (channelId: string): Promise<void> {
    let request: CloseChannelRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: CloseChannelRequest.method,
      params: [channelId]
    }
    return this.provider.ask(request).then(() => {
      return
    })
  }

  listChannels (): Promise<Array<PaymentChannel>> {
    let request: ListChannelsRequest = {
      id: randomId(),
      jsonrpc: JSONRPC,
      method: ListChannelsRequest.method,
      params: []
    }
    return this.provider.ask(request).then((response: ListChannelsResponse) => {
      return response.result.map(pc => PaymentChannelSerde.instance.deserialize(pc))
    })
  }
}
