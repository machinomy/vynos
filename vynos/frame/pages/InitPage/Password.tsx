import * as React from 'react'
import { FrameState } from '../../redux/FrameState'
import WorkerProxy from '../../WorkerProxy'
import { connect, Dispatch } from 'react-redux'
import { Button, Container, Divider, Form, Header, Icon } from 'semantic-ui-react'
import { MINIMUM_PASSWORD_LENGTH, PASSWORD_CONFIRMATION_HINT_TEXT, PASSWORD_HINT_TEXT } from '../../constants'
import RestorePage from '../RestorePage'
import Logo from '../../components/Logo'
import * as actions from '../../redux/actions'

const style = require('../../styles/ynos.css')

export interface PasswordState {
  password: string
  passwordConfirmation: string
  passwordError: null | string
  passwordConfirmationError: null | string
  displayRestore: boolean
}

export interface OwnPasswordProps {
  showVerifiable: () => void
}

export interface PasswordSubpageStateProps {
  workerProxy: WorkerProxy
}

export interface PasswordSubpageDispatchProps {
  genKeyring?: (workerProxy: WorkerProxy, password: string) => void
}

export type PasswordSubpageProps = PasswordSubpageStateProps & PasswordSubpageDispatchProps & OwnPasswordProps

export class Password extends React.Component<PasswordSubpageProps, PasswordState> {
  constructor (props: PasswordSubpageProps) {
    super(props)
    this.state = {
      password: '',
      passwordConfirmation: '',
      passwordError: null,
      passwordConfirmationError: null,
      displayRestore: false
    }
    this.handleChangePassword = this.handleChangePassword.bind(this)
    this.handleChangePasswordConfirmation = this.handleChangePasswordConfirmation.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  isValid () {
    let passwordError = this.state.passwordError
    if (this.state.password.length < MINIMUM_PASSWORD_LENGTH) {
      passwordError = PASSWORD_HINT_TEXT
      this.setState({
        passwordError: passwordError
      })
    }
    let passwordConfirmationError = this.state.passwordConfirmationError
    if (this.state.passwordConfirmation !== this.state.password && this.state.password) {
      passwordConfirmationError = PASSWORD_CONFIRMATION_HINT_TEXT
      this.setState({
        passwordConfirmationError: passwordConfirmationError
      })
    }
    return !(passwordError || passwordConfirmationError)
  }

  handleSubmit (e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault()
    if (this.isValid() && this.state.password) {
      return this.props.genKeyring!(this.props.workerProxy, this.state.password)
    }
  }

  handleChangePassword (ev: React.ChangeEvent<EventTarget>) {
    let value = (ev.target as HTMLInputElement).value
    this.setState({
      password: value,
      passwordError: null,
      passwordConfirmationError: null
    })
  }

  handleChangePasswordConfirmation (ev: React.ChangeEvent<EventTarget>) {
    let value = (ev.target as HTMLInputElement).value
    this.setState({
      passwordConfirmation: value,
      passwordError: null,
      passwordConfirmationError: null
    })
  }

  renderPasswordInput () {
    let className = this.state.passwordError ? style.inputError : ''
    return (
      <input
        type="password"
        placeholder="Password"
        className={className}
        onChange={this.handleChangePassword}
        autoComplete="new-password"
      />
    )
  }

  renderPasswordHint () {
    if (this.state.passwordError) {
      return <span className={style.errorText}><i className={style.vynosInfo}/> {this.state.passwordError}</span>
    } else {
      return <span className={style.passLenText}>At least {MINIMUM_PASSWORD_LENGTH} characters</span>
    }
  }

  renderPasswordConfirmationInput () {
    let className = this.state.passwordConfirmationError ? style.inputError : ''
    return (
      <input
        type="password"
        placeholder="Password Confirmation"
        className={className}
        onChange={this.handleChangePasswordConfirmation}
        autoComplete="new-password"
      />
    )
  }

  renderPasswordConfirmationHint () {
    if (this.state.passwordConfirmationError) {
      return <span className={style.errorText}><i className={style.vynosInfo}/> {this.state.passwordConfirmationError}</span>
    } else {
      return <span className={style.errorText}>&nbsp;</span>
    }
  }

  doDisplayRestorePage () {
    this.setState({
      displayRestore: true
    })
  }

  doneDisplayRestorePage () {
    this.setState({
      displayRestore: false
    })
  }

  render () {
    if (this.state.displayRestore) {
      return <RestorePage goBack={this.doneDisplayRestorePage.bind(this)}/>
    }

    return (
      <Container textAlign="center" className={`${style.flexContainer} ${style.clearBorder}`}>
        <Logo />
        <Divider hidden={true} />
        <Header as="h1" className={style.encryptionHeader}>Encrypt your new wallet</Header>
        <Form onSubmit={this.handleSubmit} className={style.encryptionForm}>
          <div className="equal width fields" style={{ flexDirection: 'column', textAlign: 'left' }}>
            <Form.Field className={style.clearIndent}>
              {this.renderPasswordInput()}
              {this.renderPasswordHint()}
            </Form.Field>
            <Form.Field className={style.clearIndent}>
              {this.renderPasswordConfirmationInput()}
              {this.renderPasswordConfirmationHint()}
            </Form.Field>
          </div>
          <Divider hidden={true} />
          <Button type="submit" content="Create wallet" primary={true} className={style.buttonNav} />
          <br />
          <a onClick={this.doDisplayRestorePage.bind(this)}>Restore wallet</a>
        </Form>
        <a onClick={this.props.showVerifiable} id={style.shieldIcon}><Icon name={'shield'} size={'large'}/></a>
      </Container>
    )
  }
}

function mapStateToProps (state: FrameState, props: OwnPasswordProps): PasswordSubpageStateProps & OwnPasswordProps {
  return {
    workerProxy: state.temp.workerProxy,
    showVerifiable: props.showVerifiable
  }
}

function mapDispatchToProps (dispatch: Dispatch<FrameState>): PasswordSubpageDispatchProps {
  return {
    genKeyring: (workerProxy, password) => {
      workerProxy.genKeyring(password).then(mnemonic => {
        dispatch(actions.didReceiveMnemonic(mnemonic))
      })
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Password)
