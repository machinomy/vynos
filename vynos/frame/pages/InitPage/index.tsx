import * as React from "react";
import {connect} from "react-redux";
import {FrameState} from "../../redux/FrameState";
import Terms from './Terms'
import Password from './Password'
import Mnemonic from './Mnemonic'

export interface InitPageProps {
  needAcceptTerms?: boolean
  needSetPassword?: boolean
  didGenerateMnemonic?: boolean
  mnemonic?: string|null
  showVerifiable: () => void
}

const InitPage: React.SFC<InitPageProps> = (props) => {
  if (props.needAcceptTerms) {
    return <Terms showVerifiable={props.showVerifiable} />
  } else if (props.needSetPassword) {
    return <Password showVerifiable={props.showVerifiable} />
  } else if (props.didGenerateMnemonic && props.mnemonic) {
    return <Mnemonic mnemonic={props.mnemonic} showVerifiable={props.showVerifiable} />
  } else {
    return <Terms showVerifiable={props.showVerifiable} />
  }
}

function mapStateToProps(state: FrameState, props: InitPageProps): InitPageProps {
  return {
    needAcceptTerms: !state.temp.initPage.didAcceptTerms,
    needSetPassword: !state.temp.initPage.mnemonic,
    didGenerateMnemonic: !!state.temp.initPage.mnemonic,
    mnemonic: state.temp.initPage.mnemonic,
    showVerifiable: props.showVerifiable
  }
}

export default connect<InitPageProps, undefined, any>(mapStateToProps)(InitPage)
