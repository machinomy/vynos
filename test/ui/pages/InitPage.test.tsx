import * as React from 'react'
import * as Enzyme from 'enzyme'
import { mount } from 'enzyme'
import * as EnzymeAdapter from 'enzyme-adapter-react-16'
import InitPage, { InitPageProps, mapStateToProps } from '../../../vynos/frame/pages/InitPage'
import { FrameState, initialState as frameStateInitialStateFn } from '../../../vynos/frame/redux/FrameState'
import * as redux from 'redux'
import createHashHistory from 'history/createHashHistory'
import reducers from '../../../vynos/frame/redux/reducers'
import { Provider, Store } from 'react-redux'
import { createLogger } from 'redux-logger'
import { routerMiddleware } from 'react-router-redux'
import { default as toJson } from 'enzyme-to-json'
import { Password } from '../../../vynos/frame/pages/InitPage/Password'
import * as WorkerProxy from '../../../vynos/frame/WorkerProxy'

Enzyme.configure({ adapter: new EnzymeAdapter() })

describe('<InitPage />', () => {
  let workerProxy: WorkerProxy.default
  let history
  let middleware
  let store: Store<FrameState>
  let frameState: FrameState
  let initPageProps: InitPageProps

  let spyOnGenKeyring: any

  beforeAll(() => {
    spyOnGenKeyring = jest.fn(WorkerProxy.default.prototype.genKeyring)
  })

  beforeEach(() => {
    workerProxy = new WorkerProxy.default()
    history = createHashHistory()
    middleware = redux.applyMiddleware(createLogger(), routerMiddleware(history))
    store = redux.createStore(reducers(workerProxy), frameStateInitialStateFn(workerProxy), middleware)
    frameState = frameStateInitialStateFn(workerProxy)
    initPageProps = mapStateToProps(frameState, { showVerifiable: () => {} })
  })

  it('if didAcceptTerms is not true Accept Terms page must be shown', () => {
    if (store.getState().temp.initPage.didAcceptTerms !== true) {
      const wrapper = mount(<Provider store={store} key="provider">
        <InitPage showVerifiable={initPageProps.showVerifiable}/>
      </Provider>)

      expect(toJson(wrapper)).toMatchSnapshot()
    }
  })

  it('create new wallet', (done) => {
    if (store.getState().temp.initPage.didAcceptTerms !== true) {
      const wrapper = mount(<Provider store={store} key="provider">
        <InitPage showVerifiable={initPageProps.showVerifiable}/>
      </Provider>)

      wrapper.find('button.acceptTermsButton').simulate('click')
      expect(store.getState().temp.initPage.didAcceptTerms).toBe(true)

      const wrapperPassword = mount(<Provider store={store} key="provider">
        <Password showVerifiable={initPageProps.showVerifiable} workerProxy={store.getState().temp.workerProxy}/>
      </Provider>)

      wrapperPassword.find('.newPasswordField > input').simulate('change', { target: { value: '12344321' } })
      wrapperPassword.find('.repeatNewPasswordField > input').simulate('change', { target: { value: '12344321' } })
      wrapperPassword.find('button.createWalletButton').simulate('click')
      setTimeout(() => {
        expect(spyOnGenKeyring).toHaveBeenCalledWith('12344321')
        expect(!!store.getState().temp.initPage.mnemonic).toBe(true)
        done()
      },1000)
    }
  })
})
