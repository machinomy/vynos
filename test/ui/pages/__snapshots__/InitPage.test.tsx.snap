// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InitPage /> if needAcceptTerms is true Accept Terms page must be shown 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(InitPage)
    showVerifiable={[Function]}
  >
    <InitPage
      didGenerateMnemonic={false}
      dispatch={[Function]}
      mnemonic={null}
      needAcceptTerms={true}
      needSetPassword={true}
      showVerifiable={[Function]}
    >
      <Connect(Terms)
        showVerifiable={[Function]}
      >
        <Terms
          didAcceptTerms={[Function]}
          showVerifiable={[Function]}
        >
          <Container
            className="undefined undefined"
            textAlign="center"
          >
            <div
              className="ui center aligned container undefined undefined"
            >
              <Logo>
                <div
                  style={
                    Object {
                      "width": "100px",
                    }
                  }
                >
                  <Image
                    as="img"
                    centered={true}
                    src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" x=\\"0px\\" y=\\"0px\\"
                        viewBox=\\"0 0 200 200\\" style=\\"enable-background:new 0 0 200 200;\\" xml:space=\\"preserve\\">
                    <style type=\\"text/css\\">
                    .st0{fill:#D04071;}
                    </style>
                    <g transform=\\"rotate(0 50 50)\\">
                        <path class=\\"st0\\" d=\\"M100,40L100,40c3.3,0,6,2.7,6,6v28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6V46C94,42.7,96.7,40,100,40z	\\">
                    </path>
                    </g>
                    <g transform=\\"rotate(120 50 50)\\">
                        <path class=\\"st0\\" d=\\"M68.3-78.3L68.3-78.3c3.3,0,6,2.7,6,6v28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6l0-28
                            C62.3-75.6,65-78.3,68.3-78.3z	\\">
                        </path>
                    </g>
                    <g transform=\\"rotate(240 50 50)\\">
                        <path class=\\"st0\\" d=\\"M-18.3,8.3L-18.3,8.3c3.3,0,6,2.7,6,6l0,28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6l0-28
                        C-24.3,11-21.6,8.3-18.3,8.3z	\\">
                        </path>
                    </g>
                </svg>
"
                    ui={true}
                  >
                    <img
                      className="ui centered image"
                      src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" x=\\"0px\\" y=\\"0px\\"
                        viewBox=\\"0 0 200 200\\" style=\\"enable-background:new 0 0 200 200;\\" xml:space=\\"preserve\\">
                    <style type=\\"text/css\\">
                    .st0{fill:#D04071;}
                    </style>
                    <g transform=\\"rotate(0 50 50)\\">
                        <path class=\\"st0\\" d=\\"M100,40L100,40c3.3,0,6,2.7,6,6v28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6V46C94,42.7,96.7,40,100,40z	\\">
                    </path>
                    </g>
                    <g transform=\\"rotate(120 50 50)\\">
                        <path class=\\"st0\\" d=\\"M68.3-78.3L68.3-78.3c3.3,0,6,2.7,6,6v28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6l0-28
                            C62.3-75.6,65-78.3,68.3-78.3z	\\">
                        </path>
                    </g>
                    <g transform=\\"rotate(240 50 50)\\">
                        <path class=\\"st0\\" d=\\"M-18.3,8.3L-18.3,8.3c3.3,0,6,2.7,6,6l0,28c0,3.3-2.7,6-6,6l0,0c-3.3,0-6-2.7-6-6l0-28
                        C-24.3,11-21.6,8.3-18.3,8.3z	\\">
                        </path>
                    </g>
                </svg>
"
                    />
                  </Image>
                  <h1
                    style={
                      Object {
                        "color": "#D04071",
                        "font": "400 40px 'Alegreya Sans', sans-serif",
                        "lineHeight": "0",
                      }
                    }
                  >
                    vynos
                  </h1>
                </div>
              </Logo>
              <Divider
                hidden={true}
              >
                <div
                  className="ui hidden divider"
                />
              </Divider>
              <p>
                Ready to unlock a true value
                <br />
                of quality content
                <br />
                through 
                <em>
                  real
                </em>
                 micropayments?
              </p>
              <Divider
                hidden={true}
              >
                <div
                  className="ui hidden divider"
                />
              </Divider>
              <a
                onClick={[Function]}
              />
              <Divider
                hidden={true}
              >
                <div
                  className="ui hidden divider"
                />
              </Divider>
              <Button
                as="button"
                className="undefined acceptTermsButton"
                content="Accept"
                onClick={[Function]}
                primary={true}
              >
                <button
                  className="ui primary button undefined acceptTermsButton"
                  onClick={[Function]}
                  role="button"
                >
                  Accept
                </button>
              </Button>
              <br />
              <a
                onClick={[Function]}
              >
                Restore wallet
              </a>
            </div>
          </Container>
        </Terms>
      </Connect(Terms)>
    </InitPage>
  </Connect(InitPage)>
</Provider>
`;
